File: backend/src/Domain/Entities/WorkflowRequest.cs
using AiWorkflowAutomationDashboard.Domain.Enums;

namespace AiWorkflowAutomationDashboard.Domain.Entities;

public sealed class WorkflowRequest
{
    public Guid Id { get; set; }

    public string BusinessName { get; set; } = string.Empty;

    public string Title { get; set; } = string.Empty;

    public RequestType RequestType { get; set; }

    public string Context { get; set; } = string.Empty;

    public string Notes { get; set; } = string.Empty;

    public OutputType DesiredOutputType { get; set; }

    public Priority Priority { get; set; }

    public RequestStatus Status { get; set; }

    public string? GeneratedOutput { get; set; }

    public string? ReviewedOutput { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }

    public DateTime? ProcessedAt { get; set; }

    public string? ErrorMessage { get; set; }
}

File: frontend/types/workflowRequest.ts
export type RequestType =
  | "DocumentGeneration"
  | "BusinessSummary"
  | "ClientResponse"
  | "InternalReport"
  | "ProcessAnalysis";

export type OutputType =
  | "ProfessionalEmail"
  | "StructuredReport"
  | "ActionPlan"
  | "ExecutiveSummary"
  | "ClientFacingResponse";

export type RequestStatus =
  | "Draft"
  | "Processing"
  | "Generated"
  | "Reviewed"
  | "Archived"
  | "Failed";

export type Priority = "Low" | "Medium" | "High";

export type WorkflowRequestListItem = {
  id: string;
  businessName: string;
  title: string;
  requestType: RequestType;
  desiredOutputType: OutputType;
  priority: Priority;
  status: RequestStatus;
  createdAt: string;
  updatedAt: string;
  processedAt: string | null;
};

export type WorkflowRequestDetail = WorkflowRequestListItem & {
  context: string;
  notes: string;
  generatedOutput: string | null;
  reviewedOutput: string | null;
  errorMessage: string | null;
};

export type CreateWorkflowRequestPayload = {
  businessName: string;
  title: string;
  requestType: RequestType;
  context: string;
  notes: string;
  desiredOutputType: OutputType;
  priority: Priority;
};

export type ReviewWorkflowRequestPayload = {
  reviewedOutput: string;
};

export const requestTypeOptions: RequestType[] = [
  "DocumentGeneration",
  "BusinessSummary",
  "ClientResponse",
  "InternalReport",
  "ProcessAnalysis",
];

export const outputTypeOptions: OutputType[] = [
  "ProfessionalEmail",
  "StructuredReport",
  "ActionPlan",
  "ExecutiveSummary",
  "ClientFacingResponse",
];

export const priorityOptions: Priority[] = ["Low", "Medium", "High"];
